lib/Common/Exceptions/Throw.h

```diff
@@ -137,12 +133,30 @@ namespace Js {
 
 #define CATCH_UNHANDLED_EXCEPTION(hr) \
     catch (...) \
-    {   \
-        AssertMsg(FALSE, "invalid exception thrown and didn't get handled");    \
-        hr = E_FAIL;    \
+    { \
+        AssertOrFailFastMsg(FALSE, "invalid exception thrown and didn't get handled"); \
+        hr = E_FAIL; /* Suppress C4701 */ \
     } \
     }
 
+#define THROW_KNOWN_HRESULT_EXCEPTIONS(hr, scriptContext) \
+    if (hr == E_OUTOFMEMORY) \
+    { \
+        JavascriptError::ThrowOutOfMemoryError(scriptContext); \
+    } \
+    else if (hr == VBSERR_OutOfStack) \
+    { \
+        JavascriptError::ThrowStackOverflowError(scriptContext); \
+    } \
+    else if (hr == E_ABORT) \
+    { \
+        throw Js::ScriptAbortException(); \
+    } \
+    else if (FAILED(hr)) \
```

Comment:
This need to be the inverse translate of what is caught and translated by END_TRANSLATE_KNOWN_EXCEPTION_TO_HRESULT.